disttools/setup.py

```diff
@@ -0,0 +1,129 @@
+import os
+import pathlib
+import subprocess
+from typing import List
+
+from setuptools import setup
+from wheel.bdist_wheel import bdist_wheel
+
+base_dir = pathlib.Path(__file__).parent.resolve()
+
+
+class InstallError(Exception):
+    pass
+
+
+def gprbuild(projectfile: str, project_paths: List[str] = None) -> None:
+    """
+    Run gprbuild on the given project file with library type is fixed to static PIC
+    """
+    project_paths = project_paths or []
+    env = {"GPR_PROJECT_PATH": ":".join(project_paths), **os.environ}
+    subprocess.run(
+        ["gprbuild", "-P", projectfile, "-p", "-XLIBRARY_TYPE=static-pic"],
+        check=True,
+        env=env,
+    )
+
+
+def get_gcc_location(name: str) -> str:
+    """
+    Result location of a file inside GCC installation
+    """
+    result = subprocess.check_output(["gcc", f"--print-file-name={name}"]).rstrip().decode("utf-8")
+    if result == name:
+        raise InstallError(f'File "{name}" not found')
+    return result
+
+
+class BuildParser(bdist_wheel):
+    def run(self) -> None:
+        gprbuild("gnatcoll-bindings/iconv/gnatcoll_iconv.gpr")
+        gprbuild("gnatcoll-bindings/gmp/gnatcoll_gmp.gpr")
+        gprbuild("langkit/support/langkit_support.gpr")
+        gprbuild("librflxlang.gpr", ["langkit/support"])
+        whole_archives = (
+            get_gcc_location(a) for a in ["adalib/libgnat_pic.a", "adalib/libgnarl_pic.a"]
+        )
+        archives = (
+            get_gcc_location(a)
+            for a in [
+                "gpr/static-pic/gpr/libgpr.a",
+                "gnatcoll.static-pic/libgnatcoll.a",
+            ]
+        )
+        subprocess.run(
+            [
+                "gcc",
+                "-shared",
+                "-o",
+                "python/librflxlang/librflxlang.so",
+                "-Wl,--whole-archive",
+                "lib/static-pic/dev/librflxlang.a",
+                "langkit/support/lib/static-pic/dev/liblangkit_support.a",
+                "gnatcoll-bindings/iconv/lib/static-pic/libgnatcoll_iconv.a",
+                "gnatcoll-bindings/gmp/lib/static-pic/libgnatcoll_gmp.a",
+                *whole_archives,
+                "-Wl,--no-whole-archive",
+                *archives,
+                "-lgmp",
+            ],
+            check=True,
+        )
+
+
+with open(base_dir / "README.md") as f:
+    readme = f.read()
+    setup(
+        name="RecordFlux-language",
```

Comment:
I think it would be good to do the renaming of the repository and package (#8) before we do the first release of this package to PyPI.

Reply:
Indeed. I performed the local renaming in 675836ff0b8d10783664cce139d069b3bc6cbbbf already. We'll wait with a release until the repository actually was changed. To achieve reproducibility with older RecordFlux versions I plan to

- Rename `RecordFlux-language` into `RecordFlux-parser` (using the GitHub interface)
- Create a new repository under the old name `RecordFlux-language`
- Manually push the contents of `RecordFlux-parser` into (the new) `RecordFlux-language`
- Change the README of (the new) `RecordFlux-language` to indicate it only exists for backwards compatibility and refer to `RecordFlux-parser`
- Switch (the new) `RecordFlux-language` into archived mode (using the GitHub UI)

Does that make sense? Anything missing? IMHO we could do this immediately as soon this PR has been merged...

Reply:
Renaming the repository should be possible at any time and shouldn't break anything. GitHub automatically redirects to the old repository after renaming: https://docs.github.com/en/repositories/creating-and-managing-repositories/renaming-a-repository.

Reply:
So we just rename it and rely on GitHubs redirect indefinitely?

Reply:
I think that would be fine. If we see an issue with GitHub's redirect in the future, we could still realize your suggestion.

Reply:
[Repository](https://github.com/Componolit/RecordFlux-parser) has been renamed.